### PR TITLE
mergify: support for backport-major.minor

### PR DESCRIPTION
**Pull request by v1v**
## What is the problem this PR solves?

We have agreed to use `backport-<major>.<minor>` GitHub labels rather than `backport-<major>.<minor>.0` so the same GH label strategy works for the ingest and observability projects.

I decided to keep the current minor versions with the support for both labels.
## How does this PR solve the problem?

// Explain HOW you solved the problem in your code. It is possible that during PR reviews this changes and then this section should be updated.

## Design Checklist

- [ ] I have ensured my design is stateless and will work when multiple fleet-server instances are behind a load balancer.
- [ ] I have or intend to scale test my changes, ensuring it will work reliably with 100K+ agents connected.
- [ ] I have included fail safe mechanisms to limit the load on fleet-server: rate limiting, circuit breakers, caching, load shedding, etc.

## Checklist

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/fleet-server#changelog)

### CODE REVIEW

File: .mergify.yml [hunk 1]
```diff
@@ -50,7 +50,7 @@ pull_request_rules:
     conditions:
       - merged
       - base=main
-      - label=backport-v7.17.0
+      - label ~= ^(backport-v7.17.0|backport-7.17)$
```
Comment (v1v) [inline .mergify.yml:53]:
```suggestion
      - label~=^(backport-v7.17.0|backport-7.17)$
```

File: .mergify.yml [hunk 2]
```diff
@@ -236,7 +236,7 @@ pull_request_rules:
   - name: backport patches to 8.15 branch
     conditions:
       - merged
-      - label=backport-v8.15.0
+      - label ~= ^(backport-v8.15.0|backport-8.15)$
```
Comment (v1v) [inline .mergify.yml:239]:
```suggestion
      - label~=^(backport-v8.15.0|backport-8.15)$
```

**Comment by jlind23** [comment]
buildkite test it

**Comment by v1v** [comment]
Failures are unrelated to this PR. This is about the Mergify file, there is nothing about testing the changes in the CI

**Comment by jlind23** [comment]
Force merging this as it is failing due to the lack of 9.0 snapshot version available.

**Pull request merged**